griptape/engines/rag/modules/base_rag_module.py

```diff
@@ -14,7 +15,9 @@
 
 @define(kw_only=True)
 class BaseRagModule(ABC):
-    name: str = field(default=Factory(lambda self: self.__class__.__name__, takes_self=True), kw_only=True)
+    name: str = field(
+        default=Factory(lambda self: f"{self.__class__.__name__}-{uuid.uuid4().hex}", takes_self=True), kw_only=True
+    )
```

Comment:
What does name uniqueness get us?

Reply:
Much easier to add multiple modules of the same type without having to explicitly define names. May be that's something we add to tools as well?